### PR TITLE
Nullify Vault server dependency after destroy

### PR DESCRIPTION
**Pull request by sathishavm**
Removes the Vault server foreign_key after destroying.

/cc @zendesk/samson

### Tasks
 - [ ] :+1: from team

### References
 - Jira link: https://zendesk.atlassian.net/browse/DEVEX-691

### Risks
- Level: Low

### CODE REVIEW

File: test/lib/samson/secrets/vault_server_test.rb [hunk 1]
```diff
@@ -49,6 +49,15 @@
         to_raise: Vault::HTTPError.new("address", stub(code: '200'))
       ) { refute_valid server }
     end
+
+    it "nullifies the association on destroy" do
```
Comment (grosser) [inline test/lib/samson/secrets/vault_server_test.rb]: kinda vague and focussed on the implementation

`it "removes unfindable servers from associated deploy group"`

File: test/lib/samson/secrets/vault_server_test.rb [hunk 2]
```diff
@@ -49,6 +49,15 @@
...
+      deploy_group = deploy_groups(:pod100)
+      server.deploy_groups = [deploy_group]
+      server.save
```
Comment (grosser) [inline test/lib/samson/secrets/vault_server_test.rb]: `save!` so we don't get false-positives if the `save` silently failed

File: lib/samson/secrets/vault_server.rb [hunk 3]
```diff
@@ -36,7 +36,7 @@ class VaultServer < ActiveRecord::Base
         read_timeout: 2
       }.freeze
 
-      has_many :deploy_groups
+      has_many :deploy_groups, dependent: :nullify
```
Comment (grosser) [inline lib/samson/secrets/vault_server.rb]: this is the default behavior too, so maybe that test already passes before changing the code ?

Comment (sathishavm) [reply lib/samson/secrets/vault_server.rb]: Nope, this is not the default behavior, I Just tested, It's needed.

**Pull request merged**